eclipse-scout-core/src/tabbox/SimpleTabAreaLayout.ts

```diff
@@ -99,31 +123,72 @@ export class SimpleTabAreaLayout extends AbstractLayout {
     widgets.updateFirstLastMarker(this.tabArea.getVisibleTabs());
   }
 
-  smallPrefSize(options: PrefSizeOptions & { minTabWidth?: number } = {}): Dimension {
+  smallPrefSize(options: PrefSizeOptions & { tabMinSize?: number } = {}): Dimension {
     this._initSizes();
-    options = $.extend({minTabWidth: this.tabMinWidth}, options);
+    options = $.extend({tabMinSize: this.tabMinSize}, options);
     return this.preferredLayoutSize(this.tabArea.$container, options);
   }
 
-  override preferredLayoutSize($container: JQuery, options?: PrefSizeOptions & { minTabWidth?: number }): Dimension {
+  override preferredLayoutSize($container: JQuery, options?: PrefSizeOptions & { tabMinSize?: number }): Dimension {
     this._initSizes();
-    let minTabWidth = scout.nvl(options.minTabWidth, 0) || scout.nvl(this.tabWidth, 0);
+    let getDimensionHint: (options: PrefSizeOptions) => number;
+    let createDimension: (prefSize: Dimension, minSize: number) => Dimension;
+
+    if (this._hasHorizontalTabs()) {
+      getDimensionHint = options => options.widthHint;
+      createDimension = (prefSize, minWidth) => new Dimension(minWidth, prefSize.height);
+    } else {
+      getDimensionHint = options => options.heightHint;
+      createDimension = (prefSize, minHeight) => new Dimension(prefSize.width, minHeight);
+    }
+
+    return this._preferredLayoutSize($container, options, getDimensionHint, createDimension);
+  }
+
+  protected _preferredLayoutSize($container: JQuery, options: PrefSizeOptions & { tabMinSize?: number }, getDimensionHint: (o: PrefSizeOptions) => number, createDimension: (prefSize: Dimension, minSize: number) => Dimension): Dimension {
+    let tabMinSize = scout.nvl(options.tabMinSize, 0) || scout.nvl(this.tabSize, 0);
     let numTabs = this.tabArea.getTabs().length;
-    let minWidth = numTabs * minTabWidth;
+    let minSize = numTabs * tabMinSize;
     options = $.extend({useCssSize: true}, options);
     let prefSize = graphics.prefSize(this.tabArea.$container, options);
-    if (options.widthHint && this.tabArea.displayStyle === SimpleTabArea.DisplayStyle.SPREAD_EVEN) {
-      minWidth = Math.max(options.widthHint, minWidth);
+    let sizeHint = getDimensionHint(options);
+    if (sizeHint && this.tabArea.displayStyle === SimpleTabArea.DisplayStyle.SPREAD_EVEN) {
+      minSize = Math.max(sizeHint, minSize);
     }
-    return new Dimension(minWidth, prefSize.height);
+    return createDimension(prefSize, minSize);
```

Comment:
I think its ok to inline the two variants here.

```suggestion
    if (this._horizontal) {
      return new Dimension(minSize, prefSize.height);
    }
    return new Dimension(prefSize.width, minSize);
```